src/math/p5.Vector.js

```diff
@@ -549,7 +549,9 @@ p5.Vector.prototype.dist = function dist(v) {
  * </div>
  */
 p5.Vector.prototype.normalize = function normalize() {
-  return this.mag() === 0 ? this : this.div(this.mag());
+  var len = this.mag();
+  if (len !== 0) this.mult(1 / len);
```

Comment:
great—worth a note here about the optimization, so that someone doesn't come along later and decide to replace this with div for easier reading :)

(also just noticing something weird adjacent here: `= function normalize ()`? should be just `= function ()`?)

Reply:
yeah, i seem to remember a while back someone went through and changed a bunch of methods from anonymous to named methods. something to do with not being able to see function names in the debugger.

Reply:
The discussion for named anonymous function is at #2318. Main reason is so that if that function throws an error in the console it will have a name and not just `(anonymous function)`